### PR TITLE
 Update to docker compose v2

### LINKED ISSUE #2663: Upgrade to  docker compose (v2)
## Description

See https://docs.docker.com/compose/
From June 2023, docker-compose (v1) is out of support. Mailu currently makes use of docker-compose v1. Mailu must be adapted to make use of `docker compose` (v2).

As a starter all documentation must be updated from `docker-compose` to `docker compose`.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

documentation

## What does this PR do?

Update to docker compose v2

### Related issue(s)
- close #2663

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [ ] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -156,11 +156,11 @@ Attributes explicitly specified in filters are automatically exported: there is
...
 .. code-block:: bash
 
-  $ docker-compose exec admin flask mailu config-export --output mail-config.yml
+  $ docker compose exec admin flask mailu config-export --output mail-config.yml
 
-  $ docker-compose exec admin flask mailu config-export domain.dns_mx domain.dns_spf
+  $ docker compose exec admin flask mailu config-export domain.dns_mx domain.dns_spf
 
-  $ docker-compose exec admin flask mailu config-export user.spam_threshold
+  $ docker compose exec admin flask mailu config-export user.spam_threshold
```
Comment (ghostwheel42) [inline docs/cli.rst:163]:
```suggestion
  $ docker compose exec -T admin flask mailu config-export user.email user.spam_threshold
```

File: docs/cli.rst [hunk 2]
```diff
@@ -131,7 +131,7 @@ The purpose of this command is to export the complete configuration in YAML or J
 
 .. code-block:: bash
 
-  $ docker-compose exec admin flask mailu config-export --help
+  $ docker compose exec admin flask mailu config-export --help
```
Comment (ghostwheel42) [inline docs/cli.rst:134]:
```suggestion
  $ docker compose exec -T admin flask mailu config-export --help
```

File: docs/cli.rst [hunk 3]
```diff
@@ -156,11 +156,11 @@ Attributes explicitly specified in filters are automatically exported: there is
 
 .. code-block:: bash
 
-  $ docker-compose exec admin flask mailu config-export --output mail-config.yml
+  $ docker compose exec admin flask mailu config-export --output mail-config.yml
 
-  $ docker-compose exec admin flask mailu config-export domain.dns_mx domain.dns_spf
+  $ docker compose exec admin flask mailu config-export domain.dns_mx domain.dns_spf
```
Comment (ghostwheel42) [inline docs/cli.rst:161]:
```suggestion
  $ docker compose exec -T admin flask mailu config-export domain.dns_mx domain.dns_spf
```

File: docs/cli.rst [hunk 4]
```diff
@@ -184,11 +184,11 @@ This command imports configuration data from an external YAML or JSON source.
    -n, --dry-run   Perform a trial run with no changes made.
    -?, -h, --help  Show this message and exit.
 
-The current version of docker-compose exec does not pass stdin correctly, so you have to user docker exec instead:
+The current version of docker compose exec does not pass stdin correctly, so you have to user docker exec instead:
 
 .. code-block:: bash
 
-  docker exec -i $(docker-compose ps -q admin) flask mailu config-import -nv < mail-config.yml
+  docker exec -i $(docker compose ps -q admin) flask mailu config-import -nv < mail-config.yml
```
Comment (ghostwheel42) [inline docs/cli.rst:191]:
```suggestion
  docker compose exec -T admin flask mailu config-import -nv < mail-config.yml
```

File: tests/compose/test.py [hunk 5]
```diff
@@ -88,7 +88,7 @@ def hooks():
 sys.stdout.flush()
 deadline=datetime.datetime.now()+datetime.timedelta(minutes=timeout)
 deadline=calendar.timegm(deadline.timetuple())
-print(subprocess.check_output("docker-compose -f " + compose_file + " up -d", shell=True).decode())
+print(subprocess.check_output("docker compose -f " + compose_file + " up -d", shell=True).decode())
```
Comment (ghostwheel42) [inline tests/compose/test.py:91]: I think we should start using f-strings here. 

```suggestion
print(subprocess.check_output(f"docker compose -f {compose_file} up -d", shell=True).decode())
```

File: tests/compose/test.py [hunk 6]
```diff
@@ -22,7 +22,7 @@
 def stop(exit_code):
     print_logs()
     sys.stdout.flush()
-    print(subprocess.check_output("docker-compose -f " + compose_file + " down", shell=True).decode())
+    print(subprocess.check_output("docker compose -f " + compose_file + " down", shell=True).decode())
```
Comment (ghostwheel42) [inline tests/compose/test.py:25]: I think we should start using f-strings here. 

```suggestion
    print(subprocess.check_output(f"docker compose -f {compose_file} down", shell=True).decode())
```

**Review by ghostwheel42 — CHANGES_REQUESTED** [review]

**Comment by ghostwheel42** [comment]
bors try

**Comment by Diman0** [comment]
bors try

**Comment by Diman0** [comment]
Twice the same error on the webmail test. Apparently something is different with docker compose v2 that breaks the test.
```
Running hooks
Error: No such object: webmail_webmail_1
The default page of snappymail hasn't returned 200 but 000!
Error: [ERROR]: output = , error code = 1
Printing logs ...
```

**Comment by nextgens** [comment]
bors try

**Review by Diman0 — CHANGES_REQUESTED** [review]
Found it. In docker compose V2 underscores have changed to hypens in the names. A pretty big breaking change if you ask me.

https://github.com/Mailu/Mailu/blob/0de2430868853fa7a3a6edc8d8175cb786b37d23/tests/compose/webmail/01_ensure_admin_unreachable.sh#L3
Change `webmail_webmail_1` to `webmail-webmail-1` and the test should now work fine.

**Comment by nextgens** [comment]
@Diman0 yeah, I've fixed it already :)

**Comment by Diman0** [comment]
Nvm. I did not see the PR was updated with this change.

**Review by Diman0 — APPROVED** [review]
LGTM

**Pull request merged**